libautoscoper/src/Trial.cpp

```diff
@@ -78,6 +153,62 @@ namespace xromm
     vector<string> renderResolution;
     vector<string> optimizationOffsets;
 
+    // Parse the file
+    parseFile(filename, &mayaCams, &camRootDirs, &volumeFiles, &voxelSizes, &volumeFlips, &renderResolution, &optimizationOffsets);
+
+    // Modifiy the relative paths to be absolute paths.
+    fs::path configLocation = fs::path(filename).parent_path();
+    fs::path tmpPath;
+    for (size_t i = 0; i < mayaCams.size(); ++i) {
+      tmpPath = fs::path(mayaCams[i]);
+      if (!tmpPath.is_absolute()) {
+				mayaCams[i] = (configLocation / mayaCams[i]).string();
+			}
+		}
+    for (size_t i = 0; i < camRootDirs.size(); ++i) {
+			tmpPath = fs::path(camRootDirs[i]);
+      if (!tmpPath.is_absolute()) {
+        camRootDirs[i] = (configLocation / camRootDirs[i]).string();
+      }
+		}
+    for (size_t i = 0; i < volumeFiles.size(); ++i) {
+      tmpPath = fs::path(volumeFiles[i]);
+      if (!tmpPath.is_absolute()) {
+        volumeFiles[i] = (configLocation / volumeFiles[i]).string();
+      }
+    }
+
+    // Validate the file
+    validateFile(&mayaCams, &camRootDirs, &volumeFiles, &voxelSizes, &volumeFlips, &renderResolution, &optimizationOffsets);
+
+    // Load the cameras.
+    loadCameras(&mayaCams);
+
+    // Load the volumes.
+    loadVolumes(&volumeFiles, &voxelSizes, &volumeFlips);
+
+    // Load the videos.
+    loadVideos(&camRootDirs);
+
+    // Load the offsets
+    loadOffsets(&optimizationOffsets);
+
+    // Load the render resolution
+    loadRenderResolution(&renderResolution);
+
+
+
+  }
+
+  void Trial::parseFile(const std::string& filename, std::vector<std::string>* mayaCams, std::vector<std::string>* camRootDirs, std::vector<std::string>* volumeFiles,
+    std::vector<std::string>* voxelSizes, std::vector<std::string>* volumeFlips, std::vector<std::string>* renderResolution, std::vector<std::string>* optimizationOffsets) {
```

Comment:
```suggestion
  void Trial::parseFile(const std::string& filename, std::vector<std::string>& mayaCams, std::vector<std::string>& camRootDirs, std::vector<std::string>& volumeFiles,
    std::vector<std::string>& voxelSizes, std::vector<std::string>& volumeFlips, std::vector<std::string>& renderResolution, std::vector<std::string>& optimizationOffsets) {
```

Passing by reference allows to refactor the code while minimizing the changes. It also removes the need to check if the pointer is null.